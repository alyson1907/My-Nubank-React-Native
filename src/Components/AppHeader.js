import React from 'react'
import { View, Image } from 'react-native'
import Text from './Text'
// Images/Assets
import logo from '../img/logo/Nubank_Logo.png'
import Icon from 'react-native-vector-icons/MaterialIcons'
import styles from './Styles/AppHeader'
import Colors from '../Colors'

const AppHeader = () => {
  return (
    <View style={styles.container}>
      <View style={styles.innerView}>
        <Image source={logo} />
        <Text style={styles.textName}>Alyson</Text>
      </View>
      <Icon name="keyboard-arrow-down" size={24} color={Colors.secondary} />
    </View>
  )
}

export default AppHeader
