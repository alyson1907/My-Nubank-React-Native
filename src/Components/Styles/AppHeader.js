import { StyleSheet } from 'react-native'
import Colors from '../../Colors'

const styles = StyleSheet.create({
  container: {
    paddingVertical: 22,
    alignItems: 'center'
  },

  innerView: {
    flexDirection: 'row',
    justifyContent: 'center',
  },

  textName: {
    color: Colors.font,
    fontWeight: 'bold',
    fontSize: 22,
    marginHorizontal: 8
  },

  imgArrow: {
    transform: [ { rotate: '90deg' } ],
    width: 24,
    height: 24, 
    resizeMode: 'contain'
  }
})

export default styles
