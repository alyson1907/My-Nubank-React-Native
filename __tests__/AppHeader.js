import 'react-native'
import React from 'react'
import AppHeader from '../src/Components/AppHeader.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<AppHeader />).toJSON()
  expect(tree).toMatchSnapshot()
})
