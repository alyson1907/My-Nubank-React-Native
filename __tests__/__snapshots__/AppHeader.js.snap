// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/logo/Nubank_Logo.png",
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "sans-serif-medium",
          },
          Object {
            "color": "#eee",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginHorizontal": 8,
          },
        ]
      }
    >
      Alyson
    </Text>
  </View>
  <Icon
    color="#eee"
    name="keyboard-arrow-down"
    size={24}
  />
</View>
`;
